packages/stack/blockchain/src/index.js

```diff
@@ -22,57 +21,44 @@ class Blockchain {
     embark.registerActionForEvent("pipeline:generateAll:before", this.addArtifactFile.bind(this));
 
     this.blockchainNodes = {};
-    this.events.setCommandHandler("blockchain:node:register", (clientName, startCb) => {
-      this.blockchainNodes[clientName] = startCb;
+    this.events.setCommandHandler("blockchain:node:register", (clientName, { isStartedFn, launchFn, stopFn }) => {
+
+      if (!isStartedFn) return cb(`Blockchain client '${clientName}' must be registered with an 'isStarted' function, client not registered.`);
```

Comment:
`cb` is not defined. While it could simply be added as a last argument, I saw that in `plugins/{geth,parity}/src/index.js` no callback function is passed via `events.request` so I'm not 100% sure if/how this should be refactored.

Reply:
Also noting that I rebased against `master` and fixed a missing semicolon in a `style:` commit (could be squashed in refactor).